tools/e2e-commons/helpers/utils-helper.js

```diff
@@ -232,23 +286,27 @@ export function resolveSiteUrl() {
 /**
  * Throw an error if the passed parameter is not a valid URL
  *
- * @param {string} url the string to to be validated as URL
+ * @param {string} url - the string to to be validated as URL
  */
 export function validateUrl( url ) {
-	if ( ! new URL( url ) ) {
+	const obj = new URL( url );
+	if ( ! obj ) {
```

Comment:
I don't understand what we're trying to fix here. In the old and new variation, if I pass `'asdf'` or `null` as the URL I get the following:

`Uncaught TypeError: Failed to construct 'URL': Invalid URL`

Perhaps we should wrap it in a try/catch instead. Or are we fixing something else?

Reply:
The point of this is to throw if `url` is invalid, we don't want to change that.

eslint complains about `if ( ! new URL( url ) )` with [no-constant-condition](https://eslint.org/docs/latest/rules/no-constant-condition), because `new` can never return a falsey value. Assigning it to a variable avoids that, eslint doesn't do any static analysis to realize that `obj` will never be falsey.

Changing it to just `new URL( url )` for the throwing side effect triggers [no-new](https://eslint.org/docs/latest/rules/no-new), and `const obj = new URL( url )` without the pointless `if` triggers [no-unused-vars](https://eslint.org/docs/latest/rules/no-unused-vars).

Possibly we could do
```js
if ( ! URL.canParse( url ) ) {
    throw new Error( `Undefined or invalid url!` );
}
```
but the error thrown by `new URL` may be more informative.

Reply:
👍 Got it; it was just to prevent some linting (which makes sense, as that's the point of the PR). I was looking more at the user-defined throw, and didn't see any scenario I was able to trigger it (without wrapping it in a try/catch).

In other words, if we don't want a custom throw, the `if` block shouldn't be needed at all:

```
export function validateUrl( url ) {
	const obj = new URL( url );
}
```

The two scenarios:
* the URL is valid, and nothing happens
* the URL is invalid, and it throws

Going further down the rabbit hole, I'm not even sure why we need this function in the first place if we can just use `new URL( url )`. 🤔 🤷

Not a blocker and not directly related to purpose of this PR; feel free to disregard.